client/src/components/Verify.jsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+import { useSearchParams, useNavigate } from "react-router-dom"; 
+import api from "../api/axiosConfig";
+import Button from "../utils/Button"; 
+
+function VerifyPage() {
+  const [status, setStatus] = useState("loading");
+  const [message, setMessage] = useState("");
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const token = searchParams.get("token");
+    if (!token) {
+      setStatus("error");
+      setMessage("No token provided");
+      return;
+    }
+
+    localStorage.setItem("authToken", token);
+
+    api
+      .post("/verify")
```

Comment:
We can avoid  having a direct call to our endpoint in our components. Use a service instead, check the `api/services` folder to see how to create a service. A good example is how the ping endpoint is being called and how it is being used in the Playground component. Follow that pattern. 

Create an `auth` service that will have all the other auth endpoints.